src/main/java/org/isf/medicalinventory/manager/MedicalInventoryRowManager.java

```diff
@@ -56,7 +65,17 @@ public MedicalInventoryRow newMedicalInventoryRow(MedicalInventoryRow medicalInv
 	 * @throws OHServiceException
 	 */
 	public MedicalInventoryRow updateMedicalInventoryRow(MedicalInventoryRow medicalInventoryRow) throws OHServiceException {
-		return iOoperation.updateMedicalInventoryRow(medicalInventoryRow);
+		Optional<MedicalInventoryRow> medInvRow = ioOperation.getMedicalInventoryRowById(medicalInventoryRow.getId());
+		if (medInvRow.isPresent()) {
+			MedicalInventoryRow medInvR = medInvRow.get();
+			medInvR.setLot(medicalInventoryRow.getLot());
+			medInvR.setRealqty(medicalInventoryRow.getRealQty());
+			if (medicalInventoryRow.isNewLot()) {
+				medInvR.setNewLot(true);
+			}
+			return ioOperation.updateMedicalInventoryRow(medInvR);
+		}
+		throw new OHDataValidationException(new OHExceptionMessage(MessageBundle.getMessage("angal.inventoryrow.notfound.msg")));
```

Comment:
Why is this needed? Updating an existing entity (same @Id) it should just go through without retrieving it and resetting its properties... or maybe I'm missing something?

Reply:
I will add validation method to avoid that check